fbu-app/RegisterViewController.m

```diff
@@ -0,0 +1,129 @@
+    self.registerButton.clipsToBounds = YES;
+    [self.registerButton addTarget:self action:@selector(registerUser) forControlEvents:UIControlEventTouchUpInside];
+    [self.registerButton setTitle:@"Press Me!" forState:UIControlStateNormal];
+    [self.view addSubview:self.registerButton];
+    
+    // Create label
+    self.registerLabel = [[UILabel alloc] initWithFrame:CGRectMake(37.5f, 150.0f, 300.0f, 30.0f)];
+    self.registerLabel.text = @"Register New User";
+    [[self registerLabel] setFont: [UIFont systemFontOfSize:24]];
+    self.registerLabel.textAlignment = NSTextAlignmentCenter;
+    [self.view addSubview:self.registerLabel];
+}
+
+- (void)registerUser {
+    // Initialize a user object
+    PFUser *newUser = [PFUser user];
+    
+    // Set user properties
+    newUser.username = self.usernameField.text;
+    newUser.email = self.emailField.text;
+    newUser.password = self.passwordField.text;
+    
+    // Call sign up function on user
+    [newUser signUpInBackgroundWithBlock:^(BOOL succeeded, NSError *error) {
+        if (error != nil) {
+            // Create alert to display error
+            UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"Cannot Register"
+                                                                           message:@"Invalid username, email, or password."
+                                                                    preferredStyle:(UIAlertControllerStyleAlert)];
+            // Create a try again action
+            UIAlertAction *dismissAction = [UIAlertAction actionWithTitle:@"Dismiss"
+                                                                    style:UIAlertActionStyleCancel
+                                                                  handler:^(UIAlertAction * _Nonnull action) {
+                                                                      // Handle cancel response here. Doing nothing will dismiss the view.
+                                                                  }];
+            // Add the cancel action to the alertController
+            [alert addAction:dismissAction];
+            alert.view.tintColor = [UIColor whiteColor];
+            [self presentViewController:alert animated:YES completion:nil];
+        } else {
+            // User registered successfully, automatically log in
+            [self logInUser];
+        }
+    }];
+}
+
+// Dismiss keyboard after typing
+- (BOOL)textFieldShouldReturn:(UITextField *)textField {
+    [textField resignFirstResponder];
+    return NO;
+}
+
+- (void)logInUser {
+    NSString *username = self.usernameField.text;
+    NSString *password = self.passwordField.text;
+    
+    [PFUser logInWithUsernameInBackground:username password:password block:^(PFUser * user, NSError *  error) {
+        // Display view controller that needs to be shown after successful login
+//        HomeViewController *homeVC = [[HomeViewController alloc] init];
```

Comment:
Please avoid committing in commented code. It makes things more confusing for anyone else who tries to write things in this file.